core/domain/feature_flag_services.py

```diff
@@ -0,0 +1,252 @@
+# coding: utf-8
+#
+# Copyright 2023 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The services file for the feature flag."""
+
+from __future__ import annotations
+
+import hashlib
+import secrets
+
+from core import platform_feature_list
+from core import utils
+from core.domain import caching_services
+from core.domain import feature_flag_domain
+from core.domain import feature_flag_registry as registry
+from core.platform import models
+
+from typing import Dict, List, Mapping, Optional, Set
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import config_models
+
+(config_models,) = models.Registry.import_models([models.Names.CONFIG])
+
+ALL_FEATURE_FLAGS: List[platform_feature_list.FeatureNames] = (
+    platform_feature_list.DEV_FEATURES_LIST +
+    platform_feature_list.TEST_FEATURES_LIST +
+    platform_feature_list.PROD_FEATURES_LIST
+)
+
+ALL_FEATURES_NAMES_SET: Set[str] = set(
+    feature.value for feature in ALL_FEATURE_FLAGS
+)
+
+
+class FeatureFlagNotFoundException(Exception):
+    """Exception thrown when an unknown feature flag is requested."""
+
+    pass
+
+
+def update_feature_flag(
+    feature_name: str,
+    force_enable_for_all_users: bool,
+    rollout_percentage: int,
+    user_group_ids: List[str],
+) -> None:
+    """Updates the feature flag.
+
+    Args:
+        feature_name: str. The name of the feature flag to update.
+        force_enable_for_all_users: bool. Is feature force enabled
```

Comment:
Whether the feature is force-enabled for all the users.

Reply:
Done